pkg/workceptor/kubernetes.go

```diff
@@ -651,6 +654,63 @@ func (kw *kubeUnit) runWorkUsingLogger() {
 	kw.UpdateBasicStatus(WorkStateSucceeded, "Finished", stdout.Size())
 }
 
+func shouldUseReconnect(kw *kubeUnit) bool {
+	// Attempt to detect support for streaming from pod with timestamps based on
+	// Kubernetes server version
+	// In order to use reconnect method, Kubernetes server must be at least
+	//   v1.23.14
+	//   v1.24.8
+	//   v1.25.4
+	// These versions contain a critical patch that permits connecting to the
+	// logstream with timestamps enabled.
+	// Without the patch, stdout lines would be split after 4K characters into a
+	// new line, which will cause issues in Receptor.
+	// Can override the detection by setting the RECEPTOR_KUBE_SUPPORT_RECONNECT
+	// environment variable, e.g. true, TRUE, 1, false, FALSE, 0
+	env, ok := os.LookupEnv("RECEPTOR_KUBE_SUPPORT_RECONNECT")
+	if ok {
+		envBool, err := strconv.ParseBool(env)
+		if err != nil {
+			logger.Warning("RECEPTOR_KUBE_SUPPORT_RECONNECT=%s is not a truth value. Will auto-detect reconnect support.", env)
+		} else {
+			return envBool
+		}
+	}
+	serverVerInfo, err := kw.clientset.ServerVersion()
+	if err != nil {
+		logger.Warning("could not detect Kubernetes server version, will not use reconnect support")
+
+		return false
+	}
+	semver, err := version.ParseSemantic(serverVerInfo.String())
+	if err != nil {
+		logger.Warning("could parse Kubernetes server version %s, will not use reconnect support", serverVerInfo.String())
+
+		return false
+	}
+
+	// The patch was backported to minor version 23 and 24. We must check z stream
+	// based on the minor version
+	var compatibleVer string
+	switch serverVerInfo.Minor {
```

Comment:
I might be reading ths wrong, though what happens when `serverVerInfo.Minor` is 26? is `compatibleVer` null, does `semver.AtLeast` handle that correctly?

What about `2.23.1`?

Reply:
good point, just fixed this to use a default in case we are given a minor version that is NOT 23, 24, or 25. By default we compare the server version with `v1.23.14`.

We aren't explicitly handling the Major version, but we should still get the correct return value each time.

0.23.14 - return false
2.23.14 - return true

the edge cases are really if the minor version is 24 or 25

0.24.8 - return false
2.24.8 - return true

same with 25